nncf/experimental/torch/fx/transformations.py

```diff
@@ -529,9 +583,217 @@ def fuse_conv_bn(model: torch.fx.GraphModule) -> None:
+
+    :param node: Node to get the inputs from
+    :param model: Model to which the node belongs
+    """
+    args = []
+    for arg in node.args:
+        if isinstance(arg, torch.fx.Node):
+            if arg.op != "get_attr":
+                return None
+            arg = get_tensor_constant_from_node(arg, model)
+        args.append(arg)
+    return tuple(args)
+
+
+def _compress_qdq_constant_transformation(model: torch.fx.GraphModule, matches) -> None:
+    """
+    Change the FP32 weight value to Int8 and also reshape the scale for per_channel_quantization.
+
+    :param: model: Model to apply transformations to.
+    """
+    for match in matches:
+        mul_node = match.replacements[0]
+        sub_node = match.replacements[1]
+        weight_node, scale_node, zp_node, axis = None, None, None, None
+        nodes_map = {node.name: match.nodes_map[node] for node in match.nodes_map}
+        get_const = partial(get_tensor_constant_from_node, model=model)
+        weight_node = get_const(nodes_map["weight"])
+        scale_node = get_const(nodes_map["scale"])
+        zp_node = get_const(nodes_map["zero_point"])
+        axis = nodes_map["axis"]
+        port_id = 0
+        result = None
+        if axis is not None:
+            result = torch.ops.quantized_decomposed.quantize_per_channel.default(
+                weight_node, scale_node, zp_node, axis, -128, 127, torch.int8
+            )
+            _reshape_scale_zp(model, sub_node, mul_node, weight_node, scale_node, zp_node, axis)
+        result = (
+            result
+            if result is not None
+            else torch.ops.quantized_decomposed.quantize_per_tensor.default(
+                weight_node, scale_node, zp_node, -128, 127, torch.int8
+            )
+        )
+        constant_update_fn(model, mul_node, result, port_id, updated_node_name="compressed_weight_updated_constant")
+
+
+def _reshape_scale_zp(
+    model: torch.fx.GraphModule,
+    sub_node: torch.fx.Node,
+    mul_node: torch.fx.Node,
+    weight: torch.Tensor,
+    scale: Union[torch.Tensor, float],
+    zp: Union[torch.Tensor, float, int],
+    axis: int,
+) -> None:
+    """
+    Reshape scale and zero point so that it can be multiplied elementwise with the weight for per channel quantization.
+
+    :param model: Model to apply transformations to.
+    :param node: quantize node whose scale has to be reshaped
+    """
+    new_shape = [1] * weight.dim()
+    new_shape[axis] = scale.shape[0]
+    scale = scale.reshape(new_shape)
+    zp = zp.reshape(new_shape)
+    constant_update_fn(model, sub_node, zp, 1, updated_node_name="zero_point_updated_constant")
+    constant_update_fn(model, mul_node, scale, 1, updated_node_name="scale_updated_constant")
+
+
+def fq_weights_transformation(model: torch.fx.GraphModule) -> None:
+    """
+    This function applies a transformation to replace the FP32 weights with Fake Quantized
+    FP values to avoid the rounding error when converting to OpenVino model.
```

Comment:
Unused function

Reply:
This function is used to fake quantize once so that the rounding error is reduced in OV model. It is used when we are not compressing the weights after quantization, that is, setting backend_params for `compress_weights` to false